public/static/plugins/perf/js/perf.js

```diff
@@ -463,6 +463,19 @@ mciModule.controller('PerfController', function PerfController(
 // Class to contain a collection of samples in a series.
 function TrendSamples(samples){
   this.samples = samples;
+  var NON_THREAD_LEVELS = ['start', 'end']
+  var parserMode;
+
+  // Tries to understand the data format, using inner
+  // structure differencies
+  // TODO Some better names; Need more context; A/B too abstract
+  try {
+    parserMode = samples[0].data.results[0].start === undefined
```

Comment:
I'm not a big fan of using the try/catch to handle multiple potential exceptions with guessing the data format, since if you get a different problem (ie. if results is empty) you could still choose the wrong format. Can you explicitly check for the format here?